### PR TITLE
reload scene on death

### PR DESCRIPTION
**Pull request by Rombelirk**
This PR reloads scene on character death using `SceneManager`.

### CODE REVIEW

File: Assets/Scripts/Audio/AudioManager.cs [hunk 1]
```diff
@@ -19,21 +19,16 @@ public class AudioManager : MonoBehaviour
   #region UnityMethods
   void Awake()
   {
-    if (instance != null)
```
Comment (Rombelirk) [inline Assets/Scripts/Audio/AudioManager.cs:22]: this led to all the sounds not being renewed after scene is reloaded

File: Assets/Scripts/SceneControllers/GameSceneController.cs [hunk 2]
```diff
@@ -102,23 +102,30 @@ public void OnGameOver()
...
-    gameOverUI.gameObject.SetActive(false);
-
+    
+    Scene scene = SceneManager.GetActiveScene(); 
     GameIsOver = false;
     Time.timeScale = 1;
+    SceneManager.LoadScene(scene.name);
+
+    
+    // AudioManager.instance.PlaySound("ClickSFX", false);
```
Comment (rauldel) [inline Assets/Scripts/SceneControllers/GameSceneController.cs:112]: Please, don't remove `AudioManager.instance.PlaySound("ClickSFX", false);`, we need that line for emitting a sound when clicking restart button.

Can you also remove all these commented lines?

File: Assets/Scripts/Audio/AudioManager.cs (hunk 1 above)
Comment (rauldel) [reply Assets/Scripts/Audio/AudioManager.cs:22]: Hmmm, interesting, just tried your changes and they work fine. 👍 

Can you please remove those empty lines in lines 22, 30 and 31?

**Review by rauldel — APPROVED** [review]
So far, it looks good! 🚀

File: Assets/Scripts/Audio/AudioManager.cs (hunk 1 above)
Comment (Rombelirk) [reply Assets/Scripts/Audio/AudioManager.cs:22]: sure

**Pull request merged**